src/frontend/src/scheduler/distributed/query_manager.rs

```diff
@@ -106,7 +120,31 @@ impl QueryManager {
             }
         };
 
-        query_result_fetcher.collect_rows_from_channel(format).await
+        let ret = query_result_fetcher.collect_rows_from_channel(format).await;
+
+        // Clean up queries status when ends. Note this can not be automatically done by RAII.
+        context
+            .session()
+            .env()
+            .query_manager()
+            .delete_query(&query_id);
```

Comment:
We should keep finished queries in memory for a while so that user can see metrics of finished query. We can decide the retention policy later, but at least for now we should not delete it immediately.

Reply:
Fixed